src/DataReduction/POD.jl

```diff
@@ -0,0 +1,65 @@
+function matricize(VoV::Vector{Vector{FT}}) where {FT}
+    Matrix(reduce(hcat,VoV)')
+end
+
+mutable struct POD{FT,IT} <: AbstractDRProblem
+    snapshots::Union{Vector{Vector{FT}},Matrix{FT}}
+    nmodes::IT
+    rbasis::Matrix{FT}
+    energy::FT
+
+    function POD(snaps::Vector{Vector{FT}},nmodes::IT) where {FT,IT}
+        errorhandle(matricize(snaps),nmodes)
+        new{eltype(snaps[1]),typeof(nmodes)}(snaps,nmodes,Array{FT,2}(undef,size(snaps,1),nmodes),FT(0))
+    end
+
+    function POD(snaps::Matrix{FT},nmodes::IT) where {FT,IT}
+        errorhandle(snaps,nmodes)
+        new{eltype(snaps),typeof(nmodes)}(snaps,nmodes,Array{FT,2}(undef,size(snaps,1),nmodes),FT(0))
+    end
+
+    function POD(snaps::Adjoint{FT,Matrix{FT}},nmodes::IT) where {FT,IT}
+        POD(Matrix(snaps),nmodes,Array{FT,2}(undef,size(snaps,1),nmodes),FT(0))
+    end
+end
+
+function reduce!(pod::POD{FT,IT},::SVD) where {FT,IT}
+    op_matrix = pod.snapshots
+    if typeof(pod.snapshots) == Vector{Vector{FT}}
+        op_matrix = matricize(pod.snapshots)
+    end
+    u,s,v = svd(op_matrix)
+    pod.rbasis .= u[:,1:pod.nmodes]
+    sr = s[1:pod.nmodes]
+    pod.energy = sum(sr)/sum(s)
+    nothing
+end
+
+function reduce!(pod::POD{FT,IT},::TSVD) where {FT,IT}
+    op_matrix = pod.snapshots
+    if typeof(pod.snapshots) == Vector{Vector{FT}}
+        op_matrix = matricize(pod.snapshots)
+    end
+    u,s,v = tsvd(op_matrix,pod.nmodes)
+    pod.energy = NaN
+    pod.rbasis .= u
+    nothing
+end
+
+function reduce!(pod::POD{FT,IT},::RSVD) where {FT,IT}
+    op_matrix = pod.snapshots
+    if typeof(pod.snapshots) == Vector{Vector{FT}}
+        op_matrix = matricize(pod.snapshots)
+    end
+    u,s,v = rsvd(op_matrix,pod.nmodes)
+    pod.energy = NaN
```

Comment:
It would be good to return `sum(s)/(sum(s) + (size(op_matrix,1)-pod.nmodes)*s[end])` as trivial lower bound on the rel. energy captured. In cases where using POD makes sense, that will usually be quite informative.